src/nupic/algorithms/Connections.hpp

```diff
@@ -51,9 +51,9 @@ namespace nupic
       typedef Real32 Permanence;
       typedef UInt64 Iteration;
 
-      #define CELL_MAX (USHRT_MAX-1)
-      #define SEGMENT_MAX (UCHAR_MAX-1)
-      #define SYNAPSE_MAX (UCHAR_MAX-1)
+      // Defaults
+      static const unsigned char MAX_SEGMENTS_PER_CELL = 255;
```

Comment:
Increasing this would have performance implications. How about we merge this as-is (since these are the current limits already), and open a new issue to increase these sizes and measure the performance implications?